src/util/samplebuffer.h

```diff
@@ -164,33 +163,40 @@ class SampleBuffer final {
               m_length(length) {
             DEBUG_ASSERT((buffer.size() - offset) >= length);
         }
-        CSAMPLE* data(SINT offset = 0) const {
+        [[nodiscard]] CSAMPLE* data() const noexcept {
+            return m_data;
+        }
+        [[nodiscard]] CSAMPLE* data(SINT offset) const MIXXX_RELEASE_NOEXCEPT {
             DEBUG_ASSERT((m_data != nullptr) || (offset == 0));
             DEBUG_ASSERT(0 <= offset);
             // >=: allow access to one element behind allocated memory
             DEBUG_ASSERT(m_length >= offset);
             return m_data + offset;
         }
-        SINT length(SINT offset = 0) const {
+        [[nodiscard]] SINT length() const noexcept {
+            return m_length;
+        }
+        [[nodiscard]] SINT length(SINT offset) const MIXXX_RELEASE_NOEXCEPT {
             DEBUG_ASSERT(0 <= offset);
             // >=: allow access to one element behind allocated memory
             DEBUG_ASSERT(m_length >= offset);
             return m_length - offset;
         }
-        bool empty() const {
+        [[nodiscard]] bool empty() const noexcept {
             return (m_data == nullptr) || (m_length <= 0);
         }
-        CSAMPLE& operator[](SINT index) const {
+        [[nodiscard]] CSAMPLE& operator[](SINT index) const MIXXX_RELEASE_NOEXCEPT {
             return *data(index);
         }
+
       private:
-        CSAMPLE* m_data;
-        SINT m_length;
+        CSAMPLE* m_data = nullptr;
+        SINT m_length = 0;
```

Comment:
We prefer the initialization within the constructor: 
https://github.com/mixxxdj/mixxx/wiki/Coding-Guidelines#non-static-data-member-initializers

Reply:
done